src/test/groovy/com/synacy/gradprogram/spock/exercise/ShoppingServiceSpec.groovy

```diff
@@ -0,0 +1,69 @@
+package com.synacy.gradprogram.spock.exercise
+
+import com.synacy.gradprogram.spock.demo.User
+import spock.lang.Specification
+
+class ShoppingServiceSpec extends Specification {
+
+    ShoppingService shoppingService
+    Cart cart
+    User user
+    Order order
+    OrderingService orderingService
+    DeliveryService deliveryService
+    OrderRepository orderRepository
+    DeliveryRequestRepository deliveryRequestRepository
+
+    void setup() {
+        cart = Mock(Cart)
+        user = Mock(User)
+        order = Mock(Order)
+
+        orderRepository = Mock(OrderRepository)
+        deliveryRequestRepository = Mock(DeliveryRequestRepository)
+
+        orderingService = Mock(OrderingService)
+        deliveryService = Mock(DeliveryService)
+        shoppingService = new ShoppingService(orderingService, deliveryService,
+                orderRepository, deliveryRequestRepository)
+    }
+
+    def "BuyNonSpoilingItemsInCart should check if discount applies to cart and save a delivery request"() {
```

Comment:
We should just be testing one thing per test separating the checking for the creation of delivery. We can check/assert making sure that the `orderingService.applyDiscountToCartItems(cart)` is called and remove any unnecessary stubbing or assertions.

Another thing we need to test is the creation of the order.